src/plans/conjugate_gradient_plan.jl

```diff
@@ -157,36 +174,36 @@ default vector transport and a new storage is created by default.
     > SIAM J. Optim., 10 (1999), pp. 177–182.
     > doi: [10.1137/S1052623497318992](https://doi.org/10.1137/S1052623497318992)
 """
-mutable struct DaiYuanCoefficient{TVTM<:AbstractVectorTransportMethod} <:
-               DirectionUpdateRule
+struct DaiYuanCoefficient{TVTM<:AbstractVectorTransportMethod} <: DirectionUpdateRule
     transport_method::TVTM
-    storage::StoreStateAction
-    function DaiYuanCoefficient(
-        t::AbstractVectorTransportMethod=ParallelTransport(),
-        a::StoreStateAction=StoreStateAction((:Iterate, :Gradient, :δ)),
-    )
-        return new{typeof(t)}(t, a)
+    function DaiYuanCoefficient(t::AbstractVectorTransportMethod=ParallelTransport())
```

Comment:
This one we could change to having the manifold as an optional (positional) parameter and the transport as a keyword (unified with all other places where we specify a transport) and then use the default

so
```
DaiYuanCoefficient(M::AbstractManifold=DefaultManifood(2); vector_transport_method = default_vector_transport(M))
```

because that way just providing the manifold „improves“ the default and if you want to be more precise you can also just specify the actual transport you want to use. I introduced this in a few other places as well and like this as a default scheme actually

Reply:
Sure, this makes sense. Just instead of `DefaultManifood` I will use `DefaultManifold` :wink: .

Reply:
Yes, this was written shortly before my breakfast ;)

Reply:
Done :slightly_smiling_face: